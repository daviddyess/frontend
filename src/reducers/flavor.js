import { buildActions } from 'utils';

export const types = buildActions('flavor', [
  'REQUEST_STASH',
  'REQUEST_STASH_SUCCESS',
  'REQUEST_STASH_FAILURE',
  'ADD_TO_STASH',
  'ADD_TO_STASH_SUCCESS',
  'ADD_TO_STASH_FAILURE',
  'REMOVE_FROM_STASH',
  'REMOVE_FROM_STASH_SUCCESS',
  'REMOVE_FROM_STASH_FAILURE',
  'UPDATE_STASH',
  'UPDATE_STASH_SUCCESS',
  'UPDATE_STASH_FAILURE'
]);

const requestStash = () => ({
  type: types.REQUEST_STASH
});

const requestStashSuccess = stash => ({
  type: types.REQUEST_STASH_SUCCESS,
  stash
});

const requestStashFailure = error => ({
  type: types.REQUEST_STASH_FAILURE,
  error
});

const addStash = flavor => ({
  type: types.ADD_TO_STASH,
  flavor
});

const addStashSuccess = () => ({
  type: types.ADD_TO_STASH_SUCCESS
});

const addStashFailure = error => ({
  type: types.ADD_TO_STASH_FAILURE,
  error
});

const removeStash = flavor => ({
  type: types.REMOVE_FROM_STASH,
  flavor
});

const removeStashSuccess = () => ({
  type: types.REMOVE_FROM_STASH_SUCCESS
});

const removeStashFailure = error => ({
  type: types.REMOVE_FROM_STASH_FAILURE,
  error
});

const updateStash = flavor => ({
  type: types.UPDATE_STASH,
  flavor
});

const updateStashSuccess = () => ({
  type: types.UPDATE_STASH_SUCCESS
});

const updateStashFailure = error => ({
  type: types.UPDATE_STASH_FAILURE,
  error
});

export const actions = {
  requestStash,
  requestStashSuccess,
  requestStashFailure,
  addStash,
  addStashSuccess,
  addStashFailure,
  removeStash,
  removeStashSuccess,
  removeStashFailure,
  updateStash,
  updateStashSuccess,
  updateStashFailure
};

export const initialState = {
  loaded: false,
  error: null,
  stash: []
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case types.REQUEST_STASH_SUCCESS:
      return {
        ...state,
        stash: action.stash,
        loaded: true
      };
    case types.REQUEST_STASH_FAILURE:
      return {
        ...state,
        error: action.error
      };
    case types.ADD_TO_STASH_SUCCESS:
      return {
        ...state,
        loaded: false
      };
    case types.ADD_TO_STASH_FAILURE:
      return {
        ...state,
        error: action.error
      };
    case types.REMOVE_FROM_STASH_SUCCESS:
      return {
        ...state,
        loaded: false
      };
    case types.REMOVE_FROM_STASH_FAILURE:
      return {
        ...state,
        error: action.error
      };
    case types.UPDATE_STASH_SUCCESS:
      return {
        ...state,
        loaded: false
      };
    case types.UPDATE_STASH_FAILURE:
      return {
        ...state,
        error: action.error
      };
    default:
      return state;
  }
};
