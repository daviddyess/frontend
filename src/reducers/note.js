import { buildActions } from 'utils';

export const types = buildActions('note', [
  'REQUEST_NOTE',
  'REQUEST_NOTE_SUCCESS',
  'REQUEST_NOTE_FAILURE',
  'CREATE_NOTE',
  'CREATE_NOTE_SUCCESS',
  'CREATE_NOTE_FAILURE',
  'DELETE_NOTE',
  'DELETE_NOTE_SUCCESS',
  'DELETE_NOTE_FAILURE',
  'UPDATE_NOTE',
  'UPDATE_NOTE_SUCCESS',
  'UPDATE_NOTE_FAILURE'
]);

const requestNote = note => ({
  type: types.REQUEST_NOTE,
  note
});

const requestNoteSuccess = note => ({
  type: types.REQUEST_NOTE_SUCCESS,
  note
});

const requestNoteFailure = error => ({
  type: types.REQUEST_NOTE_FAILURE,
  error
});

const createNote = flavorNote => ({
  type: types.CREATE_NOTE,
  flavorNote
});

const createNoteSuccess = () => ({
  type: types.CREATE_NOTE_SUCCESS
});

const createNoteFailure = error => ({
  type: types.CREATE_NOTE_FAILURE,
  error
});

const deleteNote = flavorNote => ({
  type: types.DELETE_NOTE,
  flavorNote
});

const deleteNoteSuccess = () => ({
  type: types.DELETE_NOTE_SUCCESS
});

const deleteNoteFailure = error => ({
  type: types.DELETE_NOTE_FAILURE,
  error
});

const updateNote = flavorNote => ({
  type: types.UPDATE_NOTE,
  flavorNote
});

const updateNoteSuccess = () => ({
  type: types.UPDATE_NOTE_SUCCESS
});

const updateNoteFailure = error => ({
  type: types.UPDATE_NOTE_FAILURE,
  error
});

export const actions = {
  requestNote,
  requestNoteSuccess,
  requestNoteFailure,
  createNote,
  createNoteSuccess,
  createNoteFailure,
  deleteNote,
  deleteNoteSuccess,
  deleteNoteFailure,
  updateNote,
  updateNoteSuccess,
  updateNoteFailure
};

export const initialState = {
  loaded: false,
  error: null,
  collection: {}
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case types.REQUEST_NOTE_SUCCESS:
      return {
        ...state,
        collection: action.note
      };
    case types.REQUEST_NOTE_FAILURE:
      return {
        ...state,
        error: action.error
      };
    case types.CREATE_NOTE_SUCCESS:
      return {
        ...state
      };
    case types.CREATE_NOTE_FAILURE:
      return {
        ...state,
        error: action.error
      };
    case types.DELETE_NOTE_SUCCESS:
      return {
        ...state,
        loaded: false
      };
    case types.DELETE_NOTE_FAILURE:
      return {
        ...state,
        error: action.error
      };
    case types.UPDATE_NOTE_SUCCESS:
      return {
        ...state,
        loaded: false
      };
    case types.UPDATE_NOTE_FAILURE:
      return {
        ...state,
        error: action.error
      };
    default:
      return state;
  }
};
