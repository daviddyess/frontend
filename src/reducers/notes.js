import { buildActions } from 'utils';

export const types = buildActions('notes', [
  'REQUEST_NOTES',
  'REQUEST_NOTES_SUCCESS',
  'REQUEST_NOTES_FAILURE'
]);

const requestNotes = notes => ({
  type: types.REQUEST_NOTES,
  notes
});

const requestNotesSuccess = notes => ({
  type: types.REQUEST_NOTES_SUCCESS,
  notes
});

const requestNotesFailure = error => ({
  type: types.REQUEST_NOTES_FAILURE,
  error
});

export const actions = {
  requestNotes,
  requestNotesSuccess,
  requestNotesFailure
};

export const initialState = {
  loaded: false,
  cache: [],
  collection: [],
  error: null
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case types.REQUEST_NOTES_SUCCESS:
      return {
        ...state,
        collection: action.notes
      };
    case types.REQUEST_NOTES_FAILURE:
      return {
        ...state,
        error: action.error
      };
    default:
      return state;
  }
};
