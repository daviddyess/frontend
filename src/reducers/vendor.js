import { buildActions } from 'utils';

export const types = buildActions('vendor', [
  'REQUEST_VENDOR',
  'REQUEST_VENDOR_SUCCESS',
  'CREATE_VENDOR',
  'DELETE_VENDOR',
  'UPDATE_VENDOR',
  'REQUEST_FAILURE'
]);

const requestVendor = (vendorId) => ({
  type: types.REQUEST_VENDOR,
  vendorId
});

const requestVendorSuccess = (vendor) => ({
  type: types.REQUEST_VENDOR_SUCCESS,
  vendor
});

const createVendor = (vendor) => ({
  type: types.CREATE_VENDOR,
  vendor
});

const deleteVendor = ({ vendorId, name }) => ({
  type: types.DELETE_VENDOR,
  vendorId,
  name
});

const updateVendor = (vendor) => ({
  type: types.UPDATE_VENDOR,
  vendor
});

const requestFailure = (error) => ({
  type: types.REQUEST_FAILURE,
  error
});

export const actions = {
  requestVendor,
  requestVendorSuccess,
  createVendor,
  deleteVendor,
  updateVendor,
  requestFailure
};

export const initialState = {
  error: null,
  collection: {}
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case types.REQUEST_VENDOR_SUCCESS:
      return {
        ...state,
        collection: action.vendor
      };
    case types.REQUEST_FAILURE:
      return {
        ...state,
        error: action.error
      };
    default:
      return state;
  }
};
