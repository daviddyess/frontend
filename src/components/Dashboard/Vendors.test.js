import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import { initialState } from 'reducers/vendors';
import { initialState as dashboardInitialState } from 'reducers/dashboard';

import ConnectedVendors, { Vendors } from './Vendors';
import { mockComponent, withMemoryRouter, withProvider } from 'utils/testing';

jest.mock('components/Pagination/Pagination', () => {
  const pagination = require('utils/testing').mockComponent('Pagination');
  const PagerInfo = require('utils/testing').mockComponent('PagerInfo');

  return {
    withPagination: () => () => pagination,
    pagination,
    PagerInfo
  };
});

describe('Dashboard <Vendors />', () => {
  const defaultLayoutOptions = {
    border: false,
    header: true,
    title: false,
    style: {}
  };
  const mockStore = configureStore();
  const store = mockStore({
    vendors: initialState,
    dashboard: dashboardInitialState
  });
  const MockPager = mockComponent('Pager');
  const props = {
    collection: [],
    pager: {},
    pagerNavigation: <MockPager />
  };

  const RoutedVendors = withMemoryRouter(ConnectedVendors);
  const DecoratedVendors = withProvider(RoutedVendors, store);

  it('renders correctly', () => {
    expect(
      renderer
        .create(<DecoratedVendors layoutOptions={defaultLayoutOptions} />)
        .toJSON()
    ).toMatchSnapshot();
  });

  it('renders instance correctly', () => {
    const component = renderer.create(
      <Vendors {...props} layoutOptions={defaultLayoutOptions} />
    );
    const { instance } = component.root.findByType(Vendors);

    expect(instance).toBeDefined();
  });
});
