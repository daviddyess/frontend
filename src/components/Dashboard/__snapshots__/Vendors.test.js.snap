// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard <Vendors /> renders correctly 1`] = `<Pagination />`;
